File: HRMS/HRMS.Web/Modules/HR/Employee/EmployeeColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;
using HRMS.Administration;

namespace HRMS.HR.Columns;

[ColumnsScript("HR.Employee")]
[BasedOnRow(typeof(EmployeeRow), CheckNames = true)]
public class EmployeeColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int EmployeeId { get; set; }
    [EditLink]
    public string EmployeeCode { get; set; }
    public string FullName { get; set; }
    public string Email { get; set; }
    public string Phone { get; set; }
    public int PaidLeavesPerMonth { get; set; }
    public string DepartmentName { get; set; }
    public string DesignationName { get; set; }
    [Width(100), AlignCenter]
    public EmployeeStatus Status { get; set; }
}

File: HRMS/HRMS.Web/Modules/ServerTypes/HR/EmployeeColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EmployeeRow } from "./EmployeeRow";
import { EmployeeStatus } from "./EmployeeStatus";

export interface EmployeeColumns {
    EmployeeId: Column<EmployeeRow>;
    EmployeeCode: Column<EmployeeRow>;
    FullName: Column<EmployeeRow>;
    Email: Column<EmployeeRow>;
    Phone: Column<EmployeeRow>;
    PaidLeavesPerMonth: Column<EmployeeRow>;
    DepartmentName: Column<EmployeeRow>;
    DesignationName: Column<EmployeeRow>;
    Status: Column<EmployeeRow>;
}

export class EmployeeColumns extends ColumnsBase<EmployeeRow> {
    static readonly columnsKey = 'HR.Employee';
    static readonly Fields = fieldsProxy<EmployeeColumns>();
}

[EmployeeStatus]; // referenced types